config/circuit-breaker-config.yml

```diff
@@ -0,0 +1,44 @@
+# Circuit breaker configurations are done at Protocol level (only http for now)
+# If an API URL is not configured in the below, then all calls to that subsystem bypass the circuit breaker.
+# One circuit is created for each of the configured API URL (Base Circuit) & additional circuits are created if configuration are overridden.
+# Appropriate circuit is chosen from bottom-up i.e. if for a particular path in API if configuration is overridden, then corresponding circuit with overriden config is applied, else Base circuit is applied)
+http:
+  error_threshold_percentage: 49 #error percentange at which the circuit trips
+  service_timeout: 180000 #timeout time by which the request must be served
+  request_volume_threshold: 10 #Number of requests post which metrics checks are applied by circuit breaker
+  sleep_window_in_ms : 120000 #ms the circuit trips before it allows for a single request to go through in half-open circuit state
+  statistical_window_length: 120000 #ms the metrics are caclucated for, after this the metrics are reset.
+  statistical_window_number_of_buckets: 10 #statistical window length is sliced into this configured number of buckets
```

Comment:
Those do not use HTTPClient, they internally use PackageCloud which internally has its own client. Defining circuit breaker for them would be a seperate excercise very similar to Riemann. In this checkin, only all those subsystems with which SF interacts directly via REST are addressed. Namely CF, BOSH, Quota Mgmt & SF itself.